src/adapter/src/client.rs

```diff
@@ -363,6 +368,44 @@ impl SessionClient {
         self.session = Some(res.session);
         res.result
     }
+
+    pub fn add_idle_in_transaction_session_timeout(&mut self) {
+        let session = self.session();
+        let timeout_dur = session.vars().idle_in_transaction_session_timeout();
+        if !timeout_dur.is_zero() {
```

Comment:
What should happen for negative durations? Probably prevented during `set`?

Reply:
Though it's also prevented during `set` because we only succeed if we parse an unsigned int.

Reply:
Just added a test for 0ms and negative ms.